src/components/Comments/Comment.js

```diff
@@ -110,6 +128,62 @@ const Comment = ({
     }
     return '#C4C4C4';
   };
+
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are two issues here:

1. Error handling: In both `onPressDM` and `onPressDMAnon`, you're catching errors and logging them, but there's no user-facing error handling. If an error occurs, the user won't know about it. Consider adding some form of user notification.

2. Loading state: You're using a single loading state object for two different actions (`loadingDm` and `loadingDmAnon`). This could lead to race conditions if both actions are triggered at the same time. Consider separating these into two different state variables.

Here's how you might address these issues:

```diff
-  const [loading, setLoading] = React.useState({
-    loadingDm: false,
-    loadingDmAnon: false
-  });
+  const [loadingDm, setLoadingDm] = React.useState(false);
+  const [loadingDmAnon, setLoadingDmAnon] = React.useState(false);

...

-  setLoading({...loading, loadingDm: true});
+  setLoadingDm(true);

...

-  console.warn(e);
+  alert('Failed to send direct message.');

...

-  setLoading({...loading, loadingDm: false});
+  setLoadingDm(false);

...

-  setLoading({...loading, loadingDmAnon: true});
+  setLoadingDmAnon(true);

...

-  console.warn(e);
+  alert('Failed to send anonymous direct message.');

...

-  setLoading({...loading, loadingDmAnon: false});
+  setLoadingDmAnon(false);

...

-  name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+  name: loadingDm ? 'Loading...' : `Message ${username}`,

...

-  name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+  name: loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->